docs/ml-features.md

```diff
@@ -1495,6 +1495,57 @@ for more details on the API.
 </div>
 
 
+## QuantileTransform
+
+QuantileTransform provide a non-parametric transformation to map the data to another
+distribution, currently both uniform and gaussian are supported.
+This model transforms the features to follow a uniform or a gaussian distribution.
```

Comment:
It looks like you copied this from scikit-learn. I wouldn't do that directly. Can we explain it more directly? It computes quantiles of each value according to its empirical CDF, then maps it back to another distribution via its inverse CDF.

For my info, why is the second step principled at all? I get operating on quantiles, but what does pretending they're quantiles of another distribution tell you? The quantiles already 'spread out' values evenly, and let you do things like compare two quantiles to see whether the distributions are similar.

Reply:
Yes, I will update this doc.

Reply:
> For my info, why is the second step principled at all? I get operating on quantiles, but what does pretending they're quantiles of another distribution tell you? The quantiles already 'spread out' values evenly, and let you do things like compare two quantiles to see whether the distributions are similar.

It can be used as a scaler, which is even more robuster than robustscaler, since it maps outliers in a relative small range.
But I use it to deal with time-series dataset, I need to make data **stationary** for following ARIMA modeling.

Reply:
Hm, I still don't really like the idea, but I accept that it exists out there. If you want to consider the top / bottom 10% of data as outliers, you already have that from quantiles, no?

For ARIMA, this isn't how you'd make the data stationary; you're removing the distribution of the data. You'd use differencing?

Reply:
Sorry for the late reply, recently I found some points on tree models seem more deserved to be improved in ml.

> For ARIMA, this isn't how you'd make the data stationary; you're removing the distribution of the data. You'd use differencing?

No, I have to take back my words.
It is often desirable to transform a time series to make it stationary. Sometimes after applying Box-Cox with a appropriate value of _lambda_ the process may _look_ stationary.
And this `QuantileTransform` can be to some degree viewed as a non-parameter variant of Box-Cox.